data/pools/transactionPool.go

```diff
@@ -366,6 +375,14 @@ func (pool *TransactionPool) OnNewBlock(block bookkeeping.Block) {
 	var knownCommitted uint
 	var unknownCommitted uint
 
+	if block.CurrentProtocol != pool.currentProtocol {
+		// invalidate caches that could depend on protocol
+		pool.lcmu.Lock()
+		pool.lsigCache.clear()
+		pool.lcmu.Unlock()
+		pool.currentProtocol = block.CurrentProtocol
```

Comment:
should this be inside the locked region? I think it might race with the check on 378 if `OnNewBlock` is called twice